FO-DICOM.Core/IO/Endian.cs

```
@@ -358,8 +358,9 @@ public class EndianBinaryReader : BinaryReader
         /// Initializes an instance of the <see cref="EndianBinaryReader"/> class.
         /// </summary>
         /// <param name="input">Stream from which to read.</param>
-        public EndianBinaryReader(Stream input)
-            : base(input)
+        /// <param name="leaveOpen"><see langword="true" /> to leave the stream open after the <see cref="T:EndianBinaryReaderr" /> object is disposed; otherwise, <see langword="false" />.</param>
+        public EndianBinaryReader(Stream input, bool leaveOpen)
+            : base(input, Encoding.UTF8, leaveOpen)
```

Maybe add a default value or an extra constructor without `leaveOpen` to avoid to have to add this parameter in all these places? I think it also doesn't help readability to have an extra `false` argument which is not self-explaining.

---

In general, I agree that boolean parameters should be avoided. But this one is a little special. The 'leaveOpen' parameter is widespread in Stream implementations in .NET, and its meaning is quite crucial: when you dispose of this thing, should it also dispose of the inner thing (leave open=false) or will you take care of it? (leave open = true)
Making the wrong decision can result in undisposed or double disposed resources. (Although .NET has a lot of built in protections against double disposal)

I am not a big fan of using a default parameter, because then you are hiding a very important decision. Generally, I find it better to be explicit and verbose about these things. 

So I think my answer is that I would prefer to keep it as is, even though your arguments are very valid.

---

That makes sense, thanks!